Enhance README to add sample data and query instructions

lakmeedee opened the pull request:
Added detailed instructions for providing own sample data and queries for northwind and woodgrove demos, including schema and policy file formats.

## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

lakmeedee committed:
Enhance README to add sample data and query instructions

Added detailed instructions for providing own sample data and queries for northwind and woodgrove demos, including schema and policy file formats.

anantshankar17 commented on big-data/README.md:
```diff
@@ -460,6 +460,86 @@ The sample query here demonstrates this [segmentedQuery.yaml](demos/analytics-ss
 
+## How do I provide my sample data and sample query?
+1. Do not run the step to generate data i.e. ./scripts/data/generate-data.ps1
+2. Add sample data for northwind to the folder big-data/demos/$demo/datasource/northwind/input. You can directly add data files or organize them in the form of folders, if required.
```
>northwind [](http://example.com/codeflow?start=23&length=9)

This readme is following a structure where the steps are written agnostic of the persona, let's follow the same.
So instead of writing the publish step for NOrthwind and WoodGrove separately, just write 1 common step for both.

anantshankar17 commented:
> # Troubleshooting

As discussed, let's also add the steps to view the stack in Grafana to check the query execution failures. This is already captured in the internal development repo.

---
Refers to: big-data/README.md:559 in 6b3563e. [](commit_id = 6b3563ed97f414caa188dbf9f0829b90ef95229c, deletion_comment = False)

anantshankar17 commented on big-data/README.md:
```diff
@@ -460,6 +460,86 @@ The sample query here demonstrates this [segmentedQuery.yaml](demos/analytics-ss
+## How do I provide my sample data and sample query?
+1. Do not run the step to generate data i.e. ./scripts/data/generate-data.ps1
+2. Add sample data for northwind to the folder big-data/demos/$demo/datasource/northwind/input. You can directly add data files or organize them in the form of folders, if required.
+3. Add the schema corresponding to the northwind data in the above folder as schema.json. The schema file should be of the following format
```
>Add the schema corresponding [](http://example.com/codeflow?start=3&length=28)

Pull the latest sample, publish-data takes the following params and defaults. Just overrides those values with your schema fields while calling publish-data.ps1.
That will make the steps much simpler than what is written right now:

These are the params with default values.
    [string]$inputSchemaFormat = "csv",
    [string]$inputSchema = "date:date,time:string,author:string,mentions:string",
    [string]$outputSchemaFormat = "csv",
    [string]$outputSchema = "author:string,Number_Of_Mentions:long,Restricted_Sum:number",
    [string]$allowedInputFields = "date,author,mentions",
    [string]$allowedOutputFields = "author,Number_Of_Mentions"

hprabh replied:
Should we also optionally take the schemaFile and policyFile as inputs instead of a string with "fieldName:type" ? We can validate that only one needs to be provided. The az cli also takes all three parameters and has such validation.

anantshankar17 replied:
Yes, that can be the second option. Deepak can share this sample with just 1 option to begin with and can add the other, later.

anantshankar17 reviewed (changes requested):
:clock1:

lakmeedee committed:
Revise demo data and query setup instructions

Updated instructions for adding sample data and queries for northwind and woodgrove demos, including schema and policy file formats.

anantshankar17 commented on big-data/README.md:
````diff
@@ -460,6 +460,22 @@ The sample query here demonstrates this [segmentedQuery.yaml](demos/analytics-ss
+   ```powershell
+      ./scripts/data/publish-data.ps1 -inputSchemaFormat <inputSchemaFormat> -inputSchema <inputSchema> -allowedInputFields <allowedInputFields> -outputSchemaFormat <outputSchemaFormat> -outputSchema <outputSchema> -allowedOutputFields <allowedOutputFields> 
+   ```
+   The allowed inputSchemaFormat and outputSchemaFormat are "csv", "json" and "parquet" <br>
````
>The allowed inputSchemaFormat and outputSchemaFormat are "csv", "json" and "parquet" <br> [](http://example.com/codeflow?start=3&length=89)

The values for these can be: ...

anantshankar17 commented on big-data/README.md:
````diff
@@ -460,6 +460,22 @@ The sample query here demonstrates this [segmentedQuery.yaml](demos/analytics-ss
+      ./scripts/data/publish-data.ps1 -inputSchemaFormat <inputSchemaFormat> -inputSchema <inputSchema> -allowedInputFields <allowedInputFields> -outputSchemaFormat <outputSchemaFormat> -outputSchema <outputSchema> -allowedOutputFields <allowedOutputFields> 
+   ```
+   The allowed inputSchemaFormat and outputSchemaFormat are "csv", "json" and "parquet" <br>
+   The inputSchema and outputSchema should be comma separated pairs of columnName:datatype like "line_item_id:long,campign:string" <br>
````
>campign [](http://example.com/codeflow?start=115&length=7)

spelling

lakmeedee committed:
Update README with corrected schema format details

Minor edits
